### PR TITLE
Add a way to open SwipeView programmatically

### LINKED ISSUE #9413: [Enhancement] Add a way to open SwipeView programatically
## Summary

In order to open the swipe menu programmatically, we need a new `Open` method. There is already [`Close`](https://github.com/xamarin/Xamarin.Forms/blob/d45bd2195063cd5064a003eedcef4caffbcea9a1/Xamarin.Forms.Core/SwipeView.cs#L62) method.

## API Changes

It could be either a new `Open` method:

    var swipeView = new SwipeView ();
    swipeView.Open(); // new API

or a new `IsOpen` property:

    var swipeView = new SwipeView ();
    swipeView.IsOpen = true; // new API

## Intended Use Case
Consider the following UI:

Screen Shot 2020-02-03 at 10 36 46 PM

To let users know that there are some buttons accessible by swipe gesture, we would also like to add a button (three dots) to open the swipe menu by tap.

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Add a way to **open** SwipeView **programmatically**.

### Issues Resolved ### 

- fixes #9413

### API Changes ###
 
 None

### Platforms Affected ### 

- Core/XAML (all platforms)
- iOS
- Android

### Behavioral/Visual Changes ###

Added:

```
public enum OpenSwipeItem
{
     LeftItems,
     TopItems,
     RightItems,
     BottomItems
}

public class SwipeView
{
     void Open(OpenSwipeItem openSwipeItem);
}
```

### Screenshots ### 

Android
fix9413-droid

iOS
fix9413

### PR Checklist ###

- [x] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by jsuarezruiz** [comment]
@jfversluis  Thanks for the feedback. There was a SwipeItems / swipe direction validation but not correct. Fixed: https://github.com/xamarin/Xamarin.Forms/pull/9635/commits/a85ef9f7ef749dffde4dda993151d651143c340b

**Comment by jfversluis** [comment]
That fixes it! Nice job!

File: Xamarin.Forms.Core/SwipeEventArgs.cs [hunk 1]
```diff
@@ -12,6 +12,16 @@ protected BaseSwipeEventArgs(SwipeDirection swipeDirection)
 		public SwipeDirection SwipeDirection { get; set; }
 	}
 
+	public class OpenSwipeEventArgs : EventArgs
+	{
+		public OpenSwipeEventArgs(OpenSwipeItem openSwipeItem)
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/SwipeEventArgs.cs]:
```suggestion
		internal OpenSwipeEventArgs(OpenSwipeItem openSwipeItem)
```

File: Xamarin.Forms.Core/SwipeView.cs [hunk 2]
```diff
@@ -57,8 +56,14 @@ public SwipeItems BottomItems
 		public event EventHandler<SwipeStartedEventArgs> SwipeStarted;
 		public event EventHandler<SwipeChangingEventArgs> SwipeChanging;
 		public event EventHandler<SwipeEndedEventArgs> SwipeEnded;
+		public event EventHandler<OpenSwipeEventArgs> OpenRequested;
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/SwipeView.cs]:
```suggestion
		[EditorBrowsable(EditorBrowsable.Never)]
		public event EventHandler<OpenSwipeEventArgs> OpenRequested;
```

**Comment by dsdilpreet** [comment]
Hey @jsuarezruiz , is there a way to reference swipe view from XAML in code behind.

In XAML
```
<SwipeView x:Name="SwipeMenu"></SwipeView>
```

In code behind
```
SwipeMenu.Open();
```

**Comment by jsuarezruiz** [comment]
Yes, you can do that.

**Comment by dsdilpreet** [comment]
@jsuarezruiz , for some reason I can't reference in my code behind. 

image

Would you have any suggestions please?

I am running
Xamarin Forms v4.6.0 pre 3
Visual Studio 16.5.2

**Comment by Only-Xam** [comment]
[#9635] Unable to open SwipeView programmatically on UWP. It is working fine on Android but not working on UWP.

Xamarin Forms Version : 4.8.0.1269

Sample Tested on UWP:
https://github.com/xamarin/xamarin-forms-samples/blob/master/UserInterface/SwipeViewDemos/SwipeViewDemos/Views/SwipeViewProgrammaticPage.xaml

**Comment by jaysonragasa** [comment]
`ctlSwipViewMenu.Open(OpenSwipeItem.RightItems);`

Showing blank, but I can tap on it.

**Pull request merged**